monitoring/mongodb/alerts.yaml

```diff
@@ -234,7 +234,11 @@ groups:
 
   - alert: MongoDbRSNotSynced
     expr: |
-      group by(rs_nm) ( count by(rs_nm, pod) (mongodb_rs_members_state{namespace="${namespace}", pod=~"${service}.*", member_state="SECONDARY"}) != (${replicas} - 1) )
+      count by (rs_nm) (
+          group by (rs_nm, member_idx) (
```

Comment:
not sure I understand how the query is different: it does the same, just computed differently...

it used for group by pod (i.e. check that every instance has a consistent view of its' replicaset), while the new query goes the other way and check that the expected number of pods are reporting the correct state of each member...

Reply:
The old query mistakenly counted how many exporter pods reported any secondaries (often resulting in 3), while the new query correctly counts the actual unique MongoDB members that are secondary (resulting in 2).
The issue was that we were comparing 2 with 3, triggering the alert...